app/controllers/experiments_controller.rb

```diff
@@ -675,9 +675,10 @@ def extract_types_for_parameters(simulation_runs)
   def extract_types_for_moes(simulation_runs)
     array_for_moes_types = []
 
-    first_run = simulation_runs.where('$and' => [{result: {'$exists' => true}}, {result: {'$ne' => nil}}]).first
+    first_run = simulation_runs.where('$and' => [{result: {'$exists' => true}}, {result: {'$ne' => {}}}, {result: {'$ne' => nil}} ]).first
```

Comment:
To query zostalo zmienione przy tym commicie bo jest bledne - nie obejmowalo przypadku pustego results. Aby zmienic na nin trzeba by test dopisac i sprawdzic czy rzeczywiscie zwraca to samo. Wydaje mi sie ze to zadanie takich refaktoringow nie obejmowalo.